File: html/examples/tutorial/example2_3.lua
require("iuplua")

function btn_exit_cb(self)
  iup.Message("Hello World Message","Hello world from IUP.")
  -- Exits the main loop
  return iup.CLOSE  
end

button = iup.button{title = "OK", action = btn_exit_cb}

vbox = iup.vbox{button}
dlg = iup.dialog{
  vbox,
  title = "Hello World 3"
}

dlg:showxy(iup.CENTER,iup.CENTER)

-- to be able to run this script inside another context
if (iup.MainLoopLevel()==0) then
  iup.MainLoop()
  iup.Close()
end

File: html/examples/tutorial/example2_3.c
#include <stdlib.h>
#include <iup.h>

int btn_exit_cb( Ihandle *self )
{
  IupMessage("Hello World Message", "Hello world from IUP.");
  /* Exits the main loop */
  return IUP_CLOSE;
}

int main(int argc, char **argv)
{
  Ihandle *dlg, *button, *vbox;

  IupOpen(&argc, &argv);
  
  button = IupButton("OK", NULL);
  vbox = IupVbox(
    button,
    NULL);
  dlg = IupDialog(vbox);
  IupSetAttribute(dlg, "TITLE", "Hello World 3");

  /* Registers callbacks */
  IupSetCallback(button, "ACTION", (Icallback) btn_exit_cb);

  IupShowXY(dlg, IUP_CENTER, IUP_CENTER);

  IupMainLoop();

  IupClose();
  return EXIT_SUCCESS;
}
